src/persist/src/s3.rs

```diff
@@ -44,17 +46,17 @@ impl S3BlobConfig {
         prefix: String,
         role_arn: Option<String>,
     ) -> Result<Self, Error> {
-        let mut config = AwsConfig::load_from_env().await;
+        let mut loader = aws_config::from_env();
```

Comment:
what are the chances that this diverges from mz_aws_util? should that expose something that AwsConfig::load_from_env and this both call?

(dunno where this comment went, i def wrote it the first time)

Reply:
About as close to 0% as you can get! `AwsConfig` is now a very simple wrapper over `aws_config::Config` that allows setting a custom global endpoint. I'm hoping that if https://github.com/awslabs/aws-sdk-rust/issues/352#issuecomment-996923751 comes true, we'll be able to delete the type entirely, and you'll just interact with `aws_config::Config` directly.

`AwsConfig::load_from_env` is just a convenience over `AwsConfig::from_loader(aws_config::from_env()).await`. If you're worried about it in the meantime, I'd be up for deleting the convenience method so that it's clear that you're meant to start with an `aws_config::Config`.

Reply:
nah, seems fine in that case. get HEAD fixed :)